pkgs/common-updater/http-updater.nix

```diff
@@ -0,0 +1,19 @@
+{ lib
+, genericUpdater
+, common-updater-scripts
+}:
+
+{ pname
+, version
+, attrPath ? pname
+, ignoredVersions ? ""
+, rev-prefix ? ""
+, odd-unstable ? false
+, patchlevel-unstable ? false
+, url ? null
+}:
+
+genericUpdater {
+  inherit pname version attrPath ignoredVersions rev-prefix odd-unstable patchlevel-unstable;
+  versionLister = "${common-updater-scripts}/bin/list-archive-versions ${lib.optionalString (url != null) "--url=${url}"}";
```

Comment:
Possibly, the URL can contain special characters that need to be escaped?

Reply:
Is there any tool in nix or nixpkgs that helps with this?

Reply:
`lib.escapeShellArg`

Reply:
I have tried that, and it does not work, because the arguments for the update script are already being quoted by `maintainers/scripts/update.nix`:
```console
--- SHOWING ERROR LOG FOR enlightenment-0.25.4 ----------------------

+ name=
+ pname=
+ version=
+ attr_path=
+ version_lister='/nix/store/lxrc27lb3ijyvaf9vwzi6dh4jpclnd93-common-updater-scripts/bin/list-directory-versions --url='\''https://download.enlightenment.org/rel/apps/enlightenment'\'''
+ ignored_versions=
+ rev_prefix=
+ odd_unstable=
+ patchlevel_unstable=
+ [[ -n '' ]]
+ name=enlightenment-0.25.4
+ [[ -n '' ]]
+ pname=enlightenment
+ [[ -n '' ]]
+ version=0.25.4
+ [[ -n '' ]]
+ attr_path=enlightenment.enlightenment
+ echo '# enlightenment-0.25.4'
++ /nix/store/lxrc27lb3ijyvaf9vwzi6dh4jpclnd93-common-updater-scripts/bin/list-directory-versions '--url='\''https://download.enlightenment.org/rel/apps/enlightenment'\''' --pname=enlightenment --attr-path=enlightenment.enlightenment --file=update-git-commits.txt
curl: (3) URL using bad/illegal format or missing URL
+ tags=
+ '[' -n '' ']'
++ echo ''
++ /nix/store/3bb3jppylvpzx2ky1p3w3cg029hkxgz1-gnugrep-3.7/bin/grep '^[0-9]'
+ tags=
```
It seems that the url gets double escaped, maybe in https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.py#L196.

Reply:
The `version_lister` assignment looks a proper escaped to me. The issue is that [this line](https://github.com/NixOS/nixpkgs/blob/a5a5849f27b5053b717bbf482f2897ded1126d5a/pkgs/common-updater/generic-updater.nix#L65) attempts to have the variable tokenized, which is unsafe. Instead it should use `sh -c "$version_lister"`.